src/hooks/AuthHooks.res

```diff
@@ -10,26 +10,44 @@ type sessionStorage = {
 external dictToObj: Js.Dict.t<'a> => {..} = "%identity"
 @val external atob: string => string = "atob"
 
-let getHeaders = (~uri, ~headers, ()) => {
+let getHeaders = (~uri, ~headers, ~xFeatureRoute, ()) => {
   let hyperSwitchToken = LocalStorage.getItem("login")->Js.Nullable.toOption
   let isMixpanel = uri->String.includes("mixpanel")
 
+  let headersForXFeature = (~token) =>
+    if (
+      uri->Js.String2.includes("lottie-files") || uri->Js.String2.includes("config/merchant-access")
+    ) {
+      headers->Dict.set("Content-Type", `application/json`)
+      headers->Dict.set("authorization", `Bearer ${token}`)
+      headers->Dict.set("api-key", `hyperswitch`)
+      headers
+    } else {
+      headers->Dict.set("authorization", `Bearer ${token}`)
+      headers->Dict.set("api-key", `hyperswitch`)
+      headers->Dict.set("x-feature", "hyperswitch-custom")
+      headers
+    }
+
   let headerObj = if isMixpanel {
     [
       ("Content-Type", "application/x-www-form-urlencoded"),
       ("accept", "application/json"),
     ]->Dict.fromArray
   } else {
-    let res = switch hyperSwitchToken {
-    | Some(token) => {
-        headers->Dict.set("authorization", `Bearer ${token}`)
-        headers->Dict.set("api-key", `hyperswitch`)
-        headers
-      }
+    let headersDict = switch hyperSwitchToken {
+    | Some(token) =>
+      xFeatureRoute
+        ? headersForXFeature(~token)
+        : {
+            headers->Dict.set("authorization", `Bearer ${token}`)
+            headers->Dict.set("api-key", `hyperswitch`)
+            headers
+          }
 
-    | None => headers
+    | None => xFeatureRoute ? headersForXFeature(~token="") : headers
```

Comment:
for xFeatureRoute the settig can be done first, then the token thing. Repetition will be reduced.